sqlc-gen-ftl/src/plugin/mod.rs

```diff
@@ -0,0 +1,187 @@
+            type_parameters: Vec::new(),
+            fields: query.params.iter().map(|param| {
+                let name = param.column.as_ref()
+                    .map(|col| col.name.clone())
+                    .unwrap_or_else(|| format!("param{}", param.number));
+                let sql_type = param.column.as_ref().and_then(|col| col.r#type.as_ref());
+                to_schema_field(name, sql_type)
+            }).collect(),
+            pos: None,
+            comments: Vec::new(),
+            metadata: Vec::new(),
+        })),
+    }
+}
+
+fn to_verb_response(query: &pluginpb::Query) -> schemapb::Decl {
+    schemapb::Decl {
+        value: Some(schemapb::decl::Value::Data(schemapb::Data {
+            name: format!("{}Response", query.name),
+            export: false,
+            type_parameters: Vec::new(),
+            fields: query.columns.iter().map(|col| {
+                to_schema_field(col.name.clone(), col.r#type.as_ref())
+            }).collect(),
+            pos: None,
+            comments: Vec::new(),
+            metadata: Vec::new(),
+        })),
+    }
+}
+
+fn to_schema_field(name: String, sql_type: Option<&pluginpb::Identifier>) -> schemapb::Field {
+    schemapb::Field {
+        name,
+        r#type: Some(sql_type.map_or_else(
+            || schemapb::Type {
+                value: Some(TypeValue::Any(schemapb::Any { pos: None })),
+            },
+            to_schema_type
+        )),
+        pos: None,
+        comments: Vec::new(),
+        metadata: Vec::new(),
+    }
+}
+
+fn to_schema_ref(module_name: &str, name: &str) -> schemapb::Type {
+    schemapb::Type {
+        value: Some(schemapb::r#type::Value::Ref(schemapb::Ref {
+            module: module_name.to_string(),
+            name: name.to_string(),
+            pos: None,
+            type_parameters: vec![],
+        }))
+    }
+}
+
+fn to_schema_type(sql_type: &pluginpb::Identifier) -> schemapb::Type {
+    let value = match sql_type.name.as_str() {
+        "integer" | "bigint" | "smallint" | "serial" | "bigserial" => 
+            TypeValue::Int(schemapb::Int { pos: None }),
+        "real" | "float" | "double" | "numeric" | "decimal" => 
+            TypeValue::Float(schemapb::Float { pos: None }),
+        "text" | "varchar" | "char" | "uuid" => 
+            TypeValue::String(schemapb::String { pos: None }),
+        "boolean" => 
+            TypeValue::Bool(schemapb::Bool { pos: None }),
+        "timestamp" | "date" | "time" => 
+            TypeValue::Time(schemapb::Time { pos: None }),
+        "json" | "jsonb" => 
+            TypeValue::Any(schemapb::Any { pos: None }),
+        "bytea" | "blob" => 
+            TypeValue::Bytes(schemapb::Bytes { pos: None }),
+        _ => 
+            TypeValue::Any(schemapb::Any { pos: None }),
+    };
+    
+    schemapb::Type {
+        value: Some(value),
+    }
+}
+
+fn get_module_name(req: &pluginpb::GenerateRequest) -> Result<String, Box<dyn std::error::Error>> {
```

Comment:
Bit weird to use `Box<dyn ...>` on an error?

Reply:
no idea what i'm doing when it comes to Rust- is the idiomatic behavior to just use `std::error::Error` directly?